### PR TITLE
Impl http mock server

### LINKED ISSUE #1: Create server to respond with mock replies
Create a mock server to respond with mock replies. This is useful creating a uniform format to mocks that will help us when we convert these to a visual format.
Another goal is to convert these to automated tests eventually.

### PR DESCRIPTION
**Pull request by manuranga**
Fixes #1

### CODE REVIEW

File: mock-http/Ballerina.toml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+[package]
+org = "baleip"
+name = "mock_http"
+version = "0.1.0"
+distribution = "2201.7.0-20230622-064700-4a2dc6dd"
```
Comment (LakshanWeerasinghe) [inline mock-http/Ballerina.toml]:
```suggestion
distribution = "2201.7.0"
```

File: mock-http/README.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+2. Add host entries to point to the mock server
+
+    eg: on *nix open `/etc/hosts` and add entry
+    ```sh
+    127.0.0.1	api.iban.com.balmock.io
+    127.0.0.1	api.intuit.com.balmock.io
+    ```
+
+3. Run the test case
```
Comment (ushirask) [inline mock-http/README.md]:
```suggestion
3. Run the service for the pattern
``` 
maybe?

File: mock-http/main.bal [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+import ballerina/io;
+
+configurable string path = ?;
+configurable int case = ?;
+
+service / on new http:Listener(80) {
+
+  resource function default [string... parm](http:Request req) returns anydata|error {
+    string host = check req.getHeader("host");
+    return check io:fileReadJson(path + "/case-" + pad2(case) + "-" + host + ".json");
```
Comment (LakshanWeerasinghe) [inline mock-http/main.bal:12]:
```suggestion
    if host.includes("hotwire") {
       return check io:fileReadXml(path + "/case-" + pad2(case) + "-" + host + ".xml");
    }
    return check io:fileReadJson(path + "/case-" + pad2(case) + "-" + host + ".json");
```

Comment (manuranga) [inline mock-http/main.bal:11]: Let's do this as a separate PR.  #12

File: mock-http/Ballerina.toml (hunk 1 above)
Comment (manuranga) [reply mock-http/Ballerina.toml]: Thanks fixed.

File: mock-http/README.md (hunk 2 above)
Comment (manuranga) [reply mock-http/README.md]: Thanks, fixed.

**Review by LakshanWeerasinghe — APPROVED** [review]
